client/state/domains/dns/actions.js

```diff
@@ -27,69 +27,59 @@ export const fetchDns = ( domainName, forceReload = false ) => ( dispatch, getSt
 
 	dispatch( { type: DOMAINS_DNS_FETCH, domainName } );
 
-	wpcom
-		.undocumented()
-		.fetchDns( domainName )
-		.then(
-			( { records } ) => dispatch( { type: DOMAINS_DNS_FETCH_COMPLETED, domainName, records } ),
-			() => dispatch( { type: DOMAINS_DNS_FETCH_FAILED, domainName } )
-		);
+	wpcom.req
+		.get( `/domains/${ domainName }/dns` )
+		.then( ( { records } ) =>
+			dispatch( { type: DOMAINS_DNS_FETCH_COMPLETED, domainName, records } )
+		)
+		.catch( () => dispatch( { type: DOMAINS_DNS_FETCH_FAILED, domainName } ) );
 };
 
+const updateDnsRequest = ( domainName, records ) =>
+	wpcom.req.post( `/domains/${ domainName }/dns`, {
+		dns: JSON.stringify( records ),
+	} );
+
 export const addDns = ( domainName, record ) => ( dispatch ) => {
 	dispatch( { type: DOMAINS_DNS_ADD, domainName, record } );
 
-	const addResult = wpcom.undocumented().updateDns( domainName, { records_to_add: [ record ] } );
-
-	addResult.then(
-		( { records } ) => dispatch( { type: DOMAINS_DNS_ADD_COMPLETED, domainName, records } ),
-		() => dispatch( { type: DOMAINS_DNS_ADD_FAILED, domainName, record } )
-	);
-
-	return addResult;
+	return updateDnsRequest( domainName, { records_to_add: [ record ] } )
+		.then( ( { records } ) => dispatch( { type: DOMAINS_DNS_ADD_COMPLETED, domainName, records } ) )
+		.catch( () => dispatch( { type: DOMAINS_DNS_ADD_FAILED, domainName, record } ) );
```

Comment:
You need to return the original `addResult` promise here, and in `deleteDns` and `applyDnsTemplate` actions, too.

The UI code does notice-displaying stuff like:
```js
dispatch( addDns() )
  .then( () => showSuccessNotice() )
  .catch( () => showErrorNotice() )
```
but here you started returning an always-resolving promise where errors are already caught.

Reply:
Good point, I've fixed that in https://github.com/Automattic/wp-calypso/pull/59056/commits/a0f174a71518523953f25aed848cca3dc9f69b8c